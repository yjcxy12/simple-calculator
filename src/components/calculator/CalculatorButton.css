.calculator-button-container {
  display: inline-block;
  margin: 1px;
}

.calculator-button-container:active {
  opacity: 0.7;
}

.calculator-button {
  background-color: transparent;
  border: 0;
  color: white;
  font-size: 30px;
  padding: 10px 20px;
  width: 100%;
}

.calculator-button:focus {
  outline: 0;
}
