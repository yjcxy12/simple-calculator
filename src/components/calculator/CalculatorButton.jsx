import React, { useCallback } from 'react';
import classNames from 'classnames';
import './CalculatorButton.css';

export const CalculatorButton = ({ className, id, label, onClick: onClickProp }) => {
  const onClick = useCallback(() => {
    onClickProp(id, label);
  }, [id, label, onClickProp]);

  return (
    <li className={classNames('calculator-button-container', className)}>
      <button data-testid={`calculator-button-${id}`} className="calculator-button" onClick={onClick} type="button">
        {label}
      </button>
    </li>
  );
};
