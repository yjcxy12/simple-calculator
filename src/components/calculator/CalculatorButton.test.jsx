import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';
import { CalculatorButton } from './CalculatorButton';

describe('snapshots', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<CalculatorButton className="test-class" id="plus" label="+" onClick={() => {}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('behaviour', () => {
  const onClick = jest.fn();
  const { queryByTestId } = render(<CalculatorButton className="test-class" id="plus" label="+" onClick={onClick} />);
  fireEvent.click(queryByTestId('calculator-button-plus'));
  expect(onClick).toHaveBeenCalledWith('plus', '+');
});
