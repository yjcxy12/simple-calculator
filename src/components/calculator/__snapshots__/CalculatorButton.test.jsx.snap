// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots renders correctly 1`] = `
<li
  className="calculator-button-container test-class"
>
  <button
    className="calculator-button"
    data-testid="calculator-button-plus"
    onClick={[Function]}
    type="button"
  >
    +
  </button>
</li>
`;
